sdk/eventhub/azure-eventhub/azure/eventhub/_client_base.py

```diff
@@ -306,7 +307,11 @@ def _create_auth(self):
                 port=self._config.connection_port,
                 verify=self._config.connection_verify
             )
-            auth.update_token()
+            # TODO: determine if below should move to `def update_token()` method
+            access_token = auth.get_token()
+            self.expires_at = access_token.expires_on
+            value = access_token.token
+            self.token = value.encode('utf-8') if isinstance(value, six.text_type) else value
```

Comment:
would the code work for authentication? I see we're setting `self.expires_at/self.token` and in which context `self` is `ClientBase`. and to your question, I think it would better be moved into `update_token` method.

btw, what I'm doing in my PR is just letting the EH Client to create the authentication object, the amqp library would handle the token for the eh client.
```
from .pyamqp.authentication import JWTTokenAuth as PyJWTTokenAuth
#...
if token_type == b"servicebus.windows.net:sastoken":
    return PyJWTTokenAuth(self._auth_uri, self._auth_uri, functools.partial(self._credential.get_token, self._auth_uri))
```

Reply:
ah maybe that's why using JWTTokenAuth wasn't working. I switched to SASTokenAuth since it wasn't working but will try this. Thanks for the snippet!